packages/flutter_tools/lib/src/android/deferred_components_setup_validator.dart

```diff
@@ -172,6 +175,7 @@ class DeferredComponentsSetupValidator {
   /// in componentB, and loading unit 4 is included in componentC.
   bool checkAppAndroidManifestComponentLoadingUnitMapping(List<DeferredComponent> components, List<LoadingUnit> generatedLoadingUnits) {
     final Directory androidDir = env.projectDir.childDirectory('android');
+    _inputs.add(env.projectDir.childFile('pubspec.yaml'));
```

Comment:
If all of these depend on the pubspec, it might be easier to list the pubspec as an input on the build rule itself, instead of adding it to the inputs potentially multiple times.

Reply:
It is technically possible for this to be run such that it does not depend on pubspec. For example, if you run only the `writeGoldens` task, then it does not depend on pubspec. However, in the current use cases, it will always depend on pubspec.